File: Aula01/Conta/conta.ts
export class Conta {
    private id: number;
    private cpf: string;
    private saldo: number;

    constructor(id: number, cpf: string, saldo: number) {
        this.id = id;
        this.cpf = cpf;
        this.saldo = saldo;
    }

    getCpf() {
        return this.cpf;
    }

    getId() {
        return this.id;
    }

    getSaldo() {
        return this.saldo;
    }

    sacar(valor: number) {
        this.saldo -= valor;
    }

    depositar(valor: number) {
        this.saldo += valor;
    }
}

function main(){
    const c1: Conta = new Conta(1, '123.456.789-00', 500.0);
    console.log('\n--------------------------------');
    mostraDadosConta(c1, "> Saldo antes do depósito: ");

    // deposito
    c1.depositar(400.0);
    mostraDadosConta(c1, "\n> Saldo após o depósito: ");

    // saque
    c1.sacar(100.0);
    mostraDadosConta(c1, "\n> Saldo após o saque: ");
    console.log('--------------------------------\n');
}

function mostraDadosConta(conta: Conta, message: string): void {
    console.log(message);
    console.log("\n| CPF do Titular: " + conta.getCpf() + "\n| Saldo: " + conta.getSaldo().toFixed(2));
}

main();

File: Aula01/Conta/Conta.java
package Conta;

public class Conta {
    private int id;
    private String cpf;
    private Double saldo;

    public Conta(){
    }

    public Conta(int id, String cpf, Double saldo){
        this.id = id;
        this.cpf = cpf;
        this.saldo = saldo;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Double getSaldo() {
        return saldo;
    }

    public String getCpf() {
        return cpf;
    }

    public void setCpf(String cpf) {
        this.cpf = cpf;
    }

    public void depositar(Double valor){
        this.saldo += valor;
    }

    public void sacar(Double valor){
        this.saldo -= valor;
    }

    public void mostraDadosConta(String message){
        System.out.println(message);
        System.out.println("\n| CPF do Titular: " + cpf + "\n| Saldo da conta: R$ " + String.format("%.2f", saldo));
    }
}
